### PR TITLE
feat: ordered list support

### PR DESCRIPTION
**Pull request by hesamurai**

### CODE REVIEW

File: src/data-engine.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ class DataEngine {
    * @param {HTMLElement} node
    * @param {object} item
    * @param {string} nodeName
-   * @returns {HTMLElement|null}
+   * @returns {Element|null}
```
Comment (ptibbetts) [inline src/data-engine.js:105]: I think the other `HTMLElement`s need changing to `Element` as well

https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

for `createItemElement` etc.

File: test/data-engine.test.js [hunk 2]
```diff
@@ -194,24 +194,24 @@ describe('DataEngine class', () => {
         data
       }
 
-      const list = document.createElement('ul')
+      const list = document.createElement('ol')
       list.innerHTML =
```
Comment (ptibbetts) [inline test/data-engine.test.js:198]: would backticks be better here so you don't have to keep writing `+` after each line?

```
list.innerHtml = `
<li data-id="1">One
  <ol data-id="1">
```

?

File: src/data-engine.js [hunk 3]
```diff
@@ -142,8 +142,8 @@ class DataEngine {
     let directParentList = this.getDirectListParentOfItem(topParent, item)
 
     if (!directParentList) {
-      // so we need to create the direct parent UL and append it to the direct parent LI
-      directParentList = this.createItemElement({ id: parent }, 'ul')
+      // so we need to create the direct parent OL and append it to the direct parent LI
```
Comment (ptibbetts) [inline src/data-engine.js:145]: personal preference I would remove the "so" from this

File: src/main.js [hunk 4]
```diff
@@ -107,14 +112,20 @@ class NestedSort {
     wrapper.appendChild(list)
   }
 
+  getListTagName() {
+    return this.listInterface === HTMLOListElement ? 'ol' : 'ul'
```
Comment (ptibbetts) [inline src/main.js:116]: is it worth checking against `HTMLUListElement` just to be sure?
or are you in control of it being `ol` or `ul` so you 100% know if it's not OL then it's UL?

File: src/main.js [hunk 5]
```diff
@@ -107,14 +112,20 @@ class NestedSort {
...
-    if (this.sortableList instanceof HTMLUListElement) return this.sortableList
+    if (this.sortableList instanceof this.listInterface) return this.sortableList
 
-    if (this.selector instanceof HTMLUListElement) {
+    if (this.selector instanceof this.listInterface) {
       this.sortableList = this.selector
     } else {
       const list = document.querySelector(this.selector)
-      this.sortableList = list.nodeName === 'UL' ? list : list.querySelector('ul')
+      this.sortableList = (list instanceof this.listInterface)
```
Comment (ptibbetts) [inline src/main.js:126]: does this need the parentheses?

**Review by ptibbetts — CHANGES_REQUESTED** [review]

File: src/data-engine.js (hunk 1 above)
Comment (hesamurai) [reply src/data-engine.js:105]: I'm afraid I cannot agree with you here, Paul. The other ones are either parameters which should be `HTMLElements`s or are returned values which would again be `HTMLElement`s.

File: test/data-engine.test.js (hunk 2 above)
Comment (hesamurai) [reply test/data-engine.test.js:198]: Not really. I tried it before. Backticks cause some white spaces which make the tests fail, unfortunately.

File: src/data-engine.js (hunk 3 above)
Comment (hesamurai) [reply src/data-engine.js]: Done! 👍🏼

File: src/main.js (hunk 5 above)
Comment (hesamurai) [reply src/main.js]: Done! 👍🏼

File: src/main.js (hunk 4 above)
Comment (hesamurai) [reply src/main.js:116]: I'd like to say that we are in control but I know that there would be some edge case like when we're dealing with a server rendered list which consists of both `ul` and `ol` lists.
I am going to create a new issue for this to implement a better error handling, maybe.

File: test/data-engine.test.js (hunk 2 above)
Comment (ptibbetts) [reply test/data-engine.test.js:198]: ah good point

File: src/data-engine.js (hunk 1 above)
Comment (ptibbetts) [reply src/data-engine.js:105]: good point, I was referring to the MDN docs for `createElement` which said `Element` but I should have clicked through where it then says `HTMLElement`

https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

> Return value
> The new Element.

https://developer.mozilla.org/en-US/docs/Web/API/Element

>  For example, the HTMLElement interface is the base interface for HTML elements,

**Review by ptibbetts — APPROVED** [review]
👍
